test/e2e_node/node_probe.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2enode
+
+import (
+	"context"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = SIGDescribe("Node Probe [NodeConformance]", func() {
+	ginkgo.Context("Kubelet", func() {
+		f := framework.NewDefaultFramework("node-probe")
+		f.NamespacePodSecurityEnforceLevel = admissionapi.LevelPrivileged
+		ginkgo.It("should run pod with readiness status set to false on termination", func() {
+			podName := "probe-test-" + string(uuid.NewUUID())
+			podClient := f.PodClient()
+			terminationGracePeriod := int64(30)
+
+			// Create Pod
+			podClient.Create(&v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: podName,
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Image:   imageutils.GetE2EImage(imageutils.Agnhost),
+							Name:    podName,
+							Command: []string{"/bin/bash", "-c", "_term() { rm -f /tmp/ready ; sleep 30; exit 0; } ; trap _term SIGTERM; touch /tmp/ready ; while true; do echo \"hello\"; sleep 10; done"},
```

Comment:
since this is long enough, it would be better to put it in heredoc form

Reply:
Fixed.